### PR TITLE
Work on subnav

### PR DESCRIPTION
**Pull request by kamiyo**
Sorry, the commit tree is kinda stupid, dunno what happened haha.

### CODE REVIEW

File: web/js/components/App/NavBar/NavBarLink.jsx [hunk 1]
```diff
@@ -2,98 +2,37 @@ import '@/less/App/NavBar/sub-nav.less';
...
+                <a onClick={() => props.toggleSub()} className={props.activeName}>
+                    <Highlight activeName={props.activeName} link={props.link} />
+                </a> :
+                <Link
+                    to={props.to}
+                    onClick={() => props.toggleSub(false)}
+                    className={props.activeName}
+                >
+                    <Highlight activeName={props.activeName} link={props.link} />
+                </Link>
```
Comment (kamiyo) [inline web/js/components/App/NavBar/NavBarLink.jsx]: Do you think this logic is okay?

File: web/js/components/App/NavBar/NavBarLinks.jsx [hunk 2]
```diff
@@ -4,20 +4,29 @@ import React from 'react';
...
-            return (
-                <NavBarLink key={i} link={link} subNavLinks={subNavLinks} showSub={props.showSub} toggleSub={props.toggleSub}/>
-            );
-        })}
+            {props.links.map(function (link, i) {
+                let activeName = '';
+                if ((props.showSub && link.path === '/media') ||
+                    (link.path === props.pathname)) {
+                    activeName = 'active';
+                }
```
Comment (kamiyo) [inline web/js/components/App/NavBar/NavBarLinks.jsx]: And this stuff?

File: web/js/components/SubNav/SubNav.jsx [hunk 3]
```diff
@@ -14,31 +14,16 @@ const Animations = {
...
-            delay: 200,
-            display: 'block'
-        });
         Velocity(subs, Animations.RightIn, {
-            delay: 250,
-            stagger: 250,
-            duration: 500,
+            delay: 100,
+            stagger: 100,
+            duration: 250,
```
Comment (kamiyo) [inline web/js/components/SubNav/SubNav.jsx]: I'm changing the animation library in another branch anyways

File: web/js/components/App/NavBar/NavBarLink.jsx [hunk 4]
```diff
@@ -2,98 +2,37 @@ import '@/less/App/NavBar/sub-nav.less';
...
-                        right: window.outerWidth - dims.left - dims.width,
-                        width: dims.width
-                    }
-                });
-            }, 100);
-
+const NavBarLink = (props) => (
+    <li className='navBarLink'>
+        {
+            (props.link === 'media') ?
```
Comment (ayc92) [inline web/js/components/App/NavBar/NavBarLink.jsx]: I think `NavBarLink` shouldn't have knowledge of the link text itself. We should instead pass in a prop called `hasSubNav` or something, and then when we loop through the links and render the `NavBarLink`s, we could do
```
links.map(link => <NavBarLink hasSubNav={props.link === 'media'} />
```
or something like that. Logically, it's the same, but I think semantically it makes more sense for this to be done outside the `NavBarLink` component.

You could even do something like
```
links.map(link => <NavBarLink hasSubNav={linksWithSubNav.includes(props.link)} />
```
where `linksWithSubNav` would be an array of strings.

**Comment by kamiyo** [comment]
@ayc92 turns out I can just use props.subNavLinks to test whether should show subnav so that was easy fix =)

**Comment by ayc92** [comment]
+1

**Pull request merged**